Sources/MongoSwift/BSON/BsonValue.swift

```diff
@@ -185,16 +167,34 @@ public struct Binary: BsonValue, Equatable, Codable {
     public let data: Data
 
     /// The binary subtype for this data.
-    public let subtype: BsonSubtype
+    public let subtype: Subtype
+
+    /// Subtypes for BSON Binary values.
+    public enum Subtype: Int, Codable {
+        /// Generic binary subtype
+        case binary = 0,
+        /// A function
+        function,
+        /// Binary (old)
+        binaryDeprecated,
+        /// UUID (old)
+        uuidDeprecated,
+        /// UUID
+        uuid,
+        /// MD5
+        md5,
+        /// User defined
+        user
+    }
 
     /// Initializes a Binary instance of the specified subtype using provided `Data`.
-    public init(data: Data, subtype: BsonSubtype) {
+    public init(data: Data, subtype: Subtype) {
```

Comment:
Also outside the scope of this PR, but you may want to validate that "data" is 16-bytes if either of the UUID sub-types is used. Some third-party tooling depends on that even though the server does not care. IMO an exception is helpful to warn users about misusing these sub-types. For instance, some were storing the string representation of a UUID (hexadecimals and dashes) instead of an actual binary value with the UUID sub-type, which is technically incorrect.

See https://jira.mongodb.org/browse/PHPC-895 and the linked GitHub issue for context.